js/samples/advanced-csharp-multiple-resources/Controllers/HomeController.cs

```diff
@@ -1,142 +1,143 @@
-		[Route("getLaunchParameters/{resourceKey}")]
+        /// <summary>
+        /// Get ImmersiveReaderLaunchParameters by using the resource that maps to the resource key
+        /// </summary>
+        /// <param name="resourceKey">The key for the resource in secrets.json</param>
+        [Route("getLaunchParameters/{resourceKey}")]
         public async Task<JsonResult> GetLaunchParameters(string resourceKey)
         {
-			if (ResourceKeyToConfigs.TryGetValue(resourceKey, out ImmersiveReaderResourceConfig resourceConfig))
-			{
-				try
-				{
-					string token = await GetTokenAsync(resourceConfig);
-
-					ImmersiveReaderLaunchParameters launchParams = new ImmersiveReaderLaunchParameters
-					{
-						Token = token,
-						Subdomain = resourceConfig.Subdomain
-					};
-
-					return new JsonResult(launchParams);
-				}
-				catch
-				{
-					return GetJsonResultError("Exception in getting token. Ensure that secrets.json is correct.", StatusCodes.Status500InternalServerError);
-				}
-			}
-			else
-			{
-				return GetJsonResultError("Invalid resource key.", StatusCodes.Status400BadRequest);
-			}
+            if (ResourceKeyToConfigs.TryGetValue(resourceKey, out ImmersiveReaderResourceConfig resourceConfig))
+            {
+                try
+                {
+                    string token = await GetTokenAsync(resourceConfig);
+
+                    ImmersiveReaderLaunchParameters launchParams = new ImmersiveReaderLaunchParameters
+                    {
+                        Token = token,
+                        Subdomain = resourceConfig.Subdomain
+                    };
+
+                    return new JsonResult(launchParams);
+                }
+                catch
+                {
+                    return GetJsonResultError("Exception in getting token. Ensure that secrets.json is correct.", StatusCodes.Status500InternalServerError);
+                }
+            }
+            else
+            {
+                return GetJsonResultError("Invalid resource key.", StatusCodes.Status400BadRequest);
+            }
         }
 
-		/// <summary>
-		/// Get an Azure AD authentication token using a given resource
-		/// </summary>
-		private async Task<string> GetTokenAsync(ImmersiveReaderResourceConfig resourceConfig)
+        /// <summary>
+        /// Get an Azure AD authentication token using a given resource
+        /// </summary>
+        private async Task<string> GetTokenAsync(ImmersiveReaderResourceConfig resourceConfig)
         {
             string authority = $"https://login.windows.net/{resourceConfig.TenantId}";
             const string resource = "https://cognitiveservices.azure.com/";
 
-            AuthenticationContext authContext = new AuthenticationContext(authority);
-            ClientCredential clientCredential = new ClientCredential(resourceConfig.ClientId, resourceConfig.ClientSecret);
+            if (app == null)
```

Comment:
add this init code to the Property

Reply:
@rwallerms 
init code added to the property

I had to move all tenant,secret, subdomain and clientId vars to class level in order to have them available for the property.